kn/fotos/entities.py

```diff
@@ -333,6 +347,42 @@ def get_random_foto_for(self, user):
                 return None
             r = 1
 
+    def search(self, q, user):
+        required_visibility = self.required_visibility(user)
+        filter={
+            'path': self.mongo_path_prefix,
+            'effectiveVisibility': {'$in': tuple(required_visibility)},
+        }
+        if q.startswith('album:'):
+            album = q[len('album:'):]
+            filter['type'] = 'album'
+            filter = {'$and': [
+                filter,
+                {'$or': [
+                    {'name': {'$regex': re.compile(re.escape(album),
+                                                   re.IGNORECASE)}},
+                    {'title': {'$regex': re.compile(re.escape(album),
+                                                    re.IGNORECASE)}}
+                ]}]}
```

Comment:
Dit gebruikt geen index voor de regex, maar MongoDB is blijkbaar slim genoeg om te zien dat er wel een index op bijvoorbeeld `album` zit waardoor hij alleen maar alle (zichtbare?) albums scant (aantal albums = 11, `nscanned`=11, `nscannedObjects`=5, `n`=1). Aangezien dit altijd via een dergelijk systeem werkte op MySQL (`LIKE %searchkey%`) verwacht ik geen performance problemen.